gluten-core/src/main/scala/io/glutenproject/expression/TernaryExpressionTransformer.scala

```diff
@@ -108,6 +108,48 @@ case class StringLocateTransformer(
   }
 }
 
+case class StringSplitTransformerBase(
+    substraitExprName: String,
+    srcExpr: ExpressionTransformer,
+    regexExpr: ExpressionTransformer,
+    limitExpr: ExpressionTransformer,
+    original: StringSplit)
+  extends ExpressionTransformer {
+
+  override def doTransform(args: java.lang.Object): ExpressionNode = {
+    TernaryExpressionTransformer(substraitExprName, srcExpr, regexExpr, limitExpr, original)
+      .doTransform(args)
+  }
+}
+
+class StringSplitTransformer(
+    substraitExprName: String,
+    srcExpr: ExpressionTransformer,
+    regexExpr: ExpressionTransformer,
+    limitExpr: ExpressionTransformer,
+    original: StringSplit)
+  extends StringSplitTransformerBase(substraitExprName, srcExpr, regexExpr, limitExpr, original)
+  with Logging {
+
+  override def doTransform(args: java.lang.Object): ExpressionNode = {
+    val limitNode = limitExpr.doTransform(args)
+    if (!limitNode.isInstanceOf[LiteralNode]) {
```

Comment:
we should also check the `regexExpr` is Literal. And also we should check that `regexExpr` should only have one char.